app/controllers/ems_infra_controller.rb

```diff
@@ -111,6 +111,48 @@ def scaledown
     end
   end
 
+  def ironic_register
```

Comment:
I am not sure if OpenStack service name (ironic) and its logic should belong to EmsInfraController. I personally would expect more generic code here.